pyccel/ast/literals.py

```diff
@@ -219,14 +219,22 @@ def python_value(self):
 
 #------------------------------------------------------------------------------
 
-class Nil(Basic, metaclass=Singleton):
+class Nil(PyccelAstNode, metaclass=Singleton):
 
     """
     class for None object in the code.
     """
-    __slots__ = ()
+    __slots__ = ('_dtype', '_precision', '_rank', '_order', '_shape')
     _attribute_nodes = ()
 
+    def __init__(self):
+        self._dtype = NativeGeneric
+        self._precision = 0
+        self._rank = 0
+        self._shape = ()
+        self._order = None
```

Comment:
See above, these can be class attributes

Reply:
Do you think `__init__` should be removed from `Nil`?

Reply:
Yes, once the object attributes are changed to class attributes it will no longer serve any purpose (I think codacy then complains)

Reply:
https://github.com/pyccel/pyccel/pull/1151/commits/1b5eeeabd856462734ac8912320e9392995723a8